vspace/src/main/java/edu/asu/diging/vspace/web/staff/StaffSpacesSearchController.java

```diff
@@ -21,27 +21,47 @@
 @Controller
 public class StaffSpacesSearchController {
 
+    private static final int PAGE_SIZE = 5;
+
     @Autowired
     private ISpaceManager spaceManager;
 
     @RequestMapping("/staff/spaces/search")
-    public ResponseEntity <String> search(@RequestParam(value = "term", required = false) String search) {
-        List <ISpace> spaces = null;
+    public ResponseEntity<String> search(
+            @RequestParam(value = "term", required = false) String search,
+            @RequestParam(value = "page", required = false, defaultValue = "1") int page) {
+
+        List<ISpace> spaces;
+        boolean hasMore;
+
         if (search != null && !search.trim().isEmpty()) {
-            spaces = spaceManager.findByName(search);
+            spaces = spaceManager.findByNamePaginated(search, page, PAGE_SIZE + 1);
```

Comment:
`Page` should have a method to get the total number pages, so this work around should not be necessary. Just get the info from the page object.